api/nodes/views.py

```diff
@@ -25,7 +26,11 @@ def get_node(self):
         obj = get_object_or_404(Node, self.kwargs[self.node_lookup_url_kwarg])
         # May raise a permission denied
         self.check_object_permissions(self.request, obj)
-        return obj
+
+        if obj.is_deleted:
```

Comment:
True that not all models will have `is_deleted` which is why you should use `getattr` or `hasattr`. 

See https://github.com/samchrisinger/osf.io/blob/bb13e189fe0739d57a4246bf9aa5b4248f210dae/framework/mongo/utils.py#L65 for an example.